packages/dds/tree/src/test/shared-tree/sharedTree.bench.ts

```diff
@@ -0,0 +1,815 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { strict as assert } from "assert";
+import { benchmark, BenchmarkType } from "@fluid-tools/benchmark";
+import { unreachableCase } from "@fluidframework/common-utils";
+import {
+	createField,
+	EditableField,
+	FieldKinds,
+	getField,
+	isEditableField,
+	isUnwrappedNode,
+	namedTreeSchema,
+	singleTextCursor,
+	UnwrappedEditableField,
+	valueSymbol,
+} from "../../feature-libraries";
+import { brand } from "../../util";
+import { ITestTreeProvider, TestTreeProvider } from "../utils";
+import { ISharedTree } from "../../shared-tree";
+import {
+	FieldKindIdentifier,
+	fieldSchema,
+	GlobalFieldKey,
+	IForestSubscription,
+	JsonableTree,
+	LocalFieldKey,
+	moveToDetachedField,
+	NamedTreeSchema,
+	rootFieldKey,
+	rootFieldKeySymbol,
+	SchemaData,
+	TreeSchemaIdentifier,
+	UpPath,
+	ValueSchema,
+} from "../../core";
+
+/**
+ * Shapes of trees to test with.
+ */
+enum TreeShape {
+	/**
+	 * Each node is inserted directly 1 layer below the root node, making a wide tree.
+	 * Used to measure the performance of editing long sequences.
+	 */
+	Wide = 0,
+	/**
+	 * Each node is inserted one layer below the previous node, making a stick-like tree.
+	 * Used to measure the performance costs related to the length of paths.
+	 */
+	Deep = 1,
+}
+
+async function createSharedTrees(
+	schemaData: SchemaData,
+	data: JsonableTree[],
+	numberOfTrees = 1,
+): Promise<readonly [ITestTreeProvider, readonly ISharedTree[]]> {
+	const provider = await TestTreeProvider.create(numberOfTrees);
+	for (const tree of provider.trees) {
+		assert(tree.isAttached());
+	}
+	provider.trees[0].storedSchema.update(schemaData);
+	provider.trees[0].context.root.insertNodes(0, data.map(singleTextCursor));
+	await provider.ensureSynchronized();
+	return [provider, provider.trees];
+}
+
+const globalFieldKey: GlobalFieldKey = brand("foo");
+const localFieldKey: LocalFieldKey = brand("foo");
+const rootSchemaName: TreeSchemaIdentifier = brand("Test");
+
+function getTestSchema(fieldKind: { identifier: FieldKindIdentifier }): SchemaData {
+	const testRootNodeSchema = namedTreeSchema({
+		name: rootSchemaName,
+		localFields: {
+			[localFieldKey]: fieldSchema(fieldKind),
+		},
+		globalFields: [globalFieldKey],
+		value: ValueSchema.Serializable,
+	});
+	const testSchemaMap: Map<TreeSchemaIdentifier, NamedTreeSchema> = new Map();
+	testSchemaMap.set(rootSchemaName, testRootNodeSchema);
+	testSchemaMap.set(dataSchema.name, dataSchema);
+	return {
+		treeSchema: testSchemaMap,
+		globalFieldSchema: new Map([
+			[rootFieldKey, fieldSchema(FieldKinds.value, [rootSchemaName])],
+		]),
+	};
+}
+
+const dataSchema = namedTreeSchema({
+	name: brand("DataSchema"),
+	localFields: {
+		foo: fieldSchema(FieldKinds.optional),
+	},
+	value: ValueSchema.Number,
+});
```

Comment:
We have a newer API for declaring schema now:


```suggestion
// Used as a linked list in "Deep" mode, and as the leaves in "Wide" mode.
const dataSchema = TypedSchema.tree("testData", {
	local: {
		foo: TypedSchema.field(FieldKinds.optional, "testData"),
	},
	value: ValueSchema.Number,
});

```
Other changes:
1. I made the type more specific, requiring the child type to be anotherTestData, I think that works.
2. I renamed the node, since dataSchema doesn't really make sense for a node type name (nodes of this type are not schema, they are test data)
3. I added documentation sating how this schema is used.

Reply:
updated schema